### PR TITLE
[Forge] Run some of the unstable tests on GCP clusters

### PR DESCRIPTION
**Pull request by sionescu**
### Description

Run part of the forge-unstable workflows on GCP, pinning to cluster "gcp-forge-0".

### Test Plan

Triggered manually: https://github.com/aptos-labs/aptos-core/actions/runs/4876839324.

### CODE REVIEW

File: .github/workflows/forge-unstable.yaml [hunk 1]
```diff
@@ -30,8 +30,11 @@ env:
   AWS_ACCOUNT_NUM: ${{ secrets.ENV_ECR_AWS_ACCOUNT_NUM }}
   AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
   AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
-  IMAGE_TAG: ${{ inputs.IMAGE_TAG }} # this is only used for workflow_dispatch, otherwise defaults to empty
   AWS_REGION: us-west-2
+  GCP_WORKLOAD_IDENTITY_PROVIDER: ${{ secrets.GCP_WORKLOAD_IDENTITY_PROVIDER }}
```
Comment (rustielin) [inline .github/workflows/forge-unstable.yaml:34]: Do you need this? 

Looks like GCP_WORKLOAD_IDENTITY_PROVIDER and other gcp auth vars are automatically pulled in in the workflow via secrets context

**Review by rustielin — DISMISSED** [review]

File: .github/workflows/forge-unstable.yaml (hunk 1 above)
Comment (sionescu) [reply .github/workflows/forge-unstable.yaml:34]: Not sure. I'll remove them to see if the workflow works without.

**Review by perryjrandall — APPROVED** [review]
instead of pinning cluster, can we just pass "gcp" or cloud flag?

**Comment by sionescu** [comment]
@perryjrandall I'll add a better form of cluster selection in another PR.

File: .github/workflows/forge-unstable.yaml (hunk 1 above)
Comment (sionescu) [reply .github/workflows/forge-unstable.yaml]: Done.

**Review by sherry-x — APPROVED** [review]
Are we just pining it to a random cluster here? wondering why some are forge 1 vs 0 or if there's any logic for selecting them here

**Comment by sionescu** [comment]
@sherry-x I added an explanation to the PR description (pin the cluster to avoid changing the global Forge config).

**Review by rustielin — APPROVED** [review]

**Pull request merged**